python-avd/pyavd/_eos_designs/structured_config/metadata/digital_twin.py

```diff
@@ -0,0 +1,40 @@
+# Copyright (c) 2023-2025 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Protocol
+
+from pyavd._errors import AristaAvdInvalidInputsError
+from pyavd._utils import default, get_v2
+
+if TYPE_CHECKING:
+    from . import AvdStructuredConfigMetadataProtocol
+
+
+class DigitalTwinMixin(Protocol):
+    """
+    Mixin Class used to generate structured config for one key.
+
+    Class should only be used as Mixin to a AvdStructuredConfig class.
+    """
+
+    def _set_digital_twin(self: AvdStructuredConfigMetadataProtocol) -> None:
+        """
+        Set the metadata for Digital Twin feature.
+
+        Only relevant to the use cases where generation of the Digital Twin infrastructure is globally enabled.
+        """
+        environment = self.inputs.digital_twin.environment
+        if digital_twin_node_type := get_v2(self.shared_utils.platform_settings.digital_twin, environment + "_node_type"):
```

Comment:
please avoid this. We loose all type checking. Do an if / else or switch and pull the relevant key.